p2p/host/autorelay/autorelay.go

```diff
@@ -135,6 +154,65 @@ func (ar *AutoRelay) background(ctx context.Context) {
 	}
 }
 
+func (ar *AutoRelay) refreshReservations(ctx context.Context, now time.Time) bool {
+	ar.mx.Lock()
+	if ar.status == network.ReachabilityPublic {
+		// we are public, forget about the relays, unprotect peers
+		for p := range ar.relays {
+			ar.host.ConnManager().Unprotect(p, autorelayTag)
+			delete(ar.relays, p)
+		}
+
+		ar.mx.Unlock()
+		return true
+	}
+
+	// find reservations about to expire and refresh them in parallel
+	var (
+		wg   sync.WaitGroup
+		fail int32
+	)
+
+	for p, rsvp := range ar.relays {
+		if rsvp == nil {
```

Comment:
Can you add a comment that this a circuitv1?

Reply:
of course.